clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -143,15 +143,29 @@ public FetchResponse(Struct struct) {
     public Send toSend(String dest, RequestHeader requestHeader) {
         ResponseHeader responseHeader = new ResponseHeader(requestHeader.correlationId());
 
+        boolean hasThrottleTime = struct.hasField(THROTTLE_TIME_KEY_NAME);
```

Comment:
I'm not necessarily opposed to using this approach to writing the fields for different versions, but I do want to call out one benefit of the previous approach that's branching on something a bit closer to `version` (and maybe that's a better way to handle this): the code for writing each version of the struct is very clear, straightforward code that just writes each of the fields, making the format very easy to compare/verify.

I didn't quickly find any examples of this in responses (maybe there are some in requests?), but I want to think about the maintainability here because once you hit ~4 versions, it becomes a lot harder to reason about what's going on with the different fields. Maybe it's fine if we have good unit test coverage, but I do worry a bit about readability.